### PR TITLE
Config file processing function for the "main" files

### PR DESCRIPTION
**Pull request by memem359**
I found this To-Do comment in the three "main" cpp files, so I made it (refactoring) happen.

`TODO Code combining config, persistent_config and commandline args is copy-pasted
slightly differently in chmain, dmain and camain.  Make it into a single function.`

I tried a single player game, and didn't notice any changes (as expected).

Since I am adding a cpp and header file to the repository, it is possible that I should have changed something else for compilation with Linux or Windows (??).
The Xcode project file is modified, for Mac OSX compilation.

### CODE REVIEW

File: server/dmain.cpp [hunk 1]
```diff
@@ -1,17 +1,11 @@
 #include "ServerApp.h"
 
+#include "../client/ConfigFileProcess.h"
```
Comment (geoffthemedio) [inline server/dmain.cpp:3]: doesn't make much sense to include a client file in the server code...

**Comment by adrianbroher** [comment]
Breaks build and includes a merge from master. Closing as invalid.  Please read the `CONTRIBUTING.md`.

**Comment by memem359** [comment]
I'll try a different PR later, but could use some advice to get it right.

adrianbroher - When you say "breaks build", is it the "merge from master", or some an additional issue? (Edit: I think I figured out what I was doing wrong with "rebase". Hopefully I won't keep making that mistake.)

geoffthemedio - Where do you suggest putting it? This is a function common to client/AI/camain, client/human/chmain, and server/dmain.

**Comment by geoffthemedio** [comment]
maybe in util?

**Pull request closed without merging**